### ISSUE TITLE
Easy onboarding for app developers, they shouldn't to build the backend

### ISSUE THREAD

**Issue by beastoin** [BasedHardware/omi#1131]
Send them:

1. API_BASE_URL: https://based-hardware-development--backend-api.modal.run/
2. Android Debug Jks, *we create one and add to firebase dev https://stackoverflow.com/questions/51845559/generate-sha-1-for-flutter-react-native-android-native-app/56091158#56091158
3. Firebase Service Account Key `firebase_dev_key.json`
4. `GoogleService-Info.plist`

That's all!

Don't forget finish prerequisites https://docs.omi.me/get_started/Setup

Command lines might help:

```
$ dart pub global activate flutterfire_cli

$ flutterfire config --platforms="android,ios" --out=lib/firebase_options_dev.dart --ios-bundle-id=com.friend-app-with-wearable.ios12.development --android-app-id=com.friend.ios.dev --android-out=android/app/src/dev/  --ios-out=ios/Config/Dev/ --ios-target="Runner" --yes --service-account="firebase_dev_key.json" --project="based-hardware-dev"

# to bypass the compile error, product only
$ flutterfire config --platforms="android,ios" --out=lib/firebase_options_prod.dart --ios-bundle-id=com.friend-app-with-wearable.ios12 --android-app-id=com.friend.ios.dev --android-out=android/app/src/prod/  --ios-out=ios/Config/Prod/ --ios-target="Runner" --yes --service-account="firebase_dev_key.json"  --project="based-hardware-dev"


```

**Comment by beastoin** [comment]
hi @adamcohenhillel feel free to ask me anything.

**Comment by josancamon19** [comment]
I think most people want to run the backend and point the app to it instead, but your call guys.

**Comment by mdmohsin7** [comment]
I agree with @josancamon19, app devs don't want to run their own backend for app specific changes and backend devs don't want to build their own app fir backend specific changes

**Comment by beastoin** [comment]
the story is, Adam called me and asked if he could help with the app developer onboarding. that's why we have this ticket, target on the app dev.

what about the backend dev? for sure bruh. we have to make their onboarding easier too. but, who want to join the ticket for backend dev ? lmk then i will take some note.

as for users? no plan for now.

**Comment by josancamon19** [comment]
And you guys sure that the developers we need are the mobile devs? not the back ones / integrations?

**Comment by beastoin** [comment]
>what about the backend dev? for sure bruh. we have to make their onboarding easier too. but, who want to join the ticket for backend dev ? lmk then i will take some note.

burhhh...

**Comment by beastoin** [comment]
hi @adamcohenhillel any question sir ?

**Comment by adamcohenhillel** [comment]
Yo @beastoin , sorry haven't seen your comments here. 

@josancamon19  - The focus on app devs is mainly to push forward the app store inside of Omi - and not for onboarding devs to help improve Omi itself - that is something separate. 


@beastoin - I understand all these steps, I did them myself, but imo this is too complicated for a 3rd party person who don't know us to follow. Like - they literally need us to add their key sha to firebase 

ANy ideas how can we simply this?

**Comment by josancamon19** [comment]
Can you show me a repo that's easy to onboard?

| @josancamon19 - The focus on app devs is mainly to push forward the app store inside of Omi 

Then you need to be able to change the back url from the app, or let the use e.g. development, which I think we already are

**Comment by josancamon19** [comment]
cc https://github.com/orgs/BasedHardware/projects/1/views/1?pane=issue&itemId=85289036&issue=BasedHardware%7Comi%7C1212

**Comment by beastoin** [comment]
>they literally need us to add their key sha to firebase

@adamcohenhillel we do this step, just send them jks file. i just updated the description to clarify it. for short: we give them the api path + 3 files, that's all.

**Comment by kodjima33** [comment]
@beastoin no one will manually send anything to the devs

only if this will be publicly accessible than it might work

I've removed @adamcohenhillel from this task as seems he is busy with his own app rn

@beastoin we need someone to be assigned here asap

**Comment by beastoin** [comment]
>publicly accessible 

ofc man

**Comment by adamcohenhillel** [comment]
Sure Nik feel free to remove me - all up to you.

The solution I came up with to make it a bit simpler is:
Create a public dev public keystore, and place it in the repo together with the rest of the code.

This will allow us to have 1 dev keystore we can sign and add its sha256 ans sha1 to the Firebase project
+ adjust the Flutter config files to point to that by default

that way we are not blocking devs to jump on the repo

We potentially can do something similar with the other 2 files

**Comment by adamcohenhillel** [comment]
Feel free to let someone else implement this.

Either way - it needs to come with a FULL youtube video of 5 mins on how to setup local frontend app and develop your first app with Omi

A few more words in docs wont do

**Comment by mdmohsin7** [comment]
In the long run, we should move authentication to the backend so that there's no dependency on Firebase in the app (except for notifications). This would also make things easy for backend devs as well (they can just simply set their backend URL in the prod app)

**Comment by adamcohenhillel** [comment]
@mdmohsin7 you'd still need furebase auth in that case^

**Comment by mdmohsin7** [comment]
> @mdmohsin7 you'd still need furebase auth in that case^

For the app?

**Comment by beastoin** [comment]
@mdmohsin7 do you want to join this task ? if yes, pls finish in-progress tickets then move to this task. if no, i will find someone else but next week.

**Comment by mdmohsin7** [comment]
My Proposal:

For short-term:
- Create a new branch `dev`, in that branch include the dev firebase files alongside a dummy debug jks
- All the changes and commits/PRs will happen on the dev branch, and during release those changes will be merged in the main branch
- Users won't have to do any setup to get the app running without backend (they'll only have to add the endpoint url in env)
- Does exposing the json file and jks harms us? The firebase project will be on free plan, so shouldn't be a big issue and the jks will be a dummy one which is solely created for this purpose

For long-term:
- Move the complete auth logic to backend (google and apple will give us the token on the front, we send it to backend and authenticate the user with Firebase and once done we create a custom jwt token)
- We will store the jwt token on the front and with every API call, user will send this. If it is expired, we will refresh it
- This way the only dependency on Firebase on the frontend will be for notifications
